### DISCUSSION TITLE
Behavior of BuildUtilities.isCICS (and others) method

### DISCUSSION THREAD

**Discussion by mhardin18** [IBM/dbb-zappbuild#553] [General]
I came across an issue today with a client where the LogicalFile.isCICS() method is correctly returning true based on what I know it to scan for. However, this client needed the member to not be recognized as a CICS program which this flag drives the behavior of in zApp. 
(Don't ask me the specifics of why this needed to happen, I'm not a CICS programmer)

So my first thought was to just use the file property override setting isCICS to false. But this does not work.

Looking at the code
`
 * isCICS - tests to see if the program is a CICS program. If the logical file is false, then
 * check to see if there is a file property.
 */
def isCICS(LogicalFile logicalFile) {
	boolean isCICS = logicalFile.isCICS()
	if (!isCICS) {
		String cicsFlag = props.getFileProperty('isCICS', logicalFile.getFile())
		if (cicsFlag)
			isCICS = cicsFlag.toBoolean()
	}

	return isCICS
}
` 

This method is only one directional. You can only override the property isCICS to be true if the LogicalFile returns false.

Why is this and the other is??? methods only one directional and not bidirectional?

**Comment by mhardin18** [comment]
My short term solution was to rewrite the method to allow for bidirectional property override. 
I could have probably solved the issue with a language definition as well, setting whatever needed to be done for the compile. But overriding the property is much simpler and in this case, seems warranted.

However, depending on the answer to this, I may change my approach.

@dennis-behm 
@M-DLB

**Comment by dennis-behm** [comment]
Hi @mhardin18 , sorry in the delay to get back to your comment/question.

It's hard to tell why it is like it is, because it is like it from the very beginning (I know not very convincing).

These runtime flags, that can be overridden via the file properties, are just to turn things `on`, but not `off`. That's for these cases, when scanners don't automatically detect the flags by scanning the source.

What is the particular scenario, where you want to turn it a flag `off`, so to `false`? It sounds odd to turn off the CICS options for a CICS source ... Overriding to `false` via properties obviously can lead to some layer-8 issues, and it could loose the dynamic evaluation that comes with DBB (to turn it on). That's why I believe, the scanner results have a higher priority.

Feels like an edge case, that you worked around.
